### PR TITLE
Stop radio music when snatched

### LINKED ISSUE #112: trolley music radio doesn't stop playing music if snatched
the radio doesn't stop playing music if you snatch it, even when the trolley starts moving

### PR DESCRIPTION
**Pull request by Commenter25**
Closes #112 

Self-explanatory. I'm not sure how necessary the whole double-radio hack is anymore, but when I tried to revert it, I just couldn't steal the radio at all anymore, so I've left that as it is.

### CODE REVIEW

**Comment by SageJFox** [comment]
Y'know, I was debating doing this when I updated the radio in cedf69993f9c1ad11078ddb4a7e92c04dbf60f57. (Speaking of which, the double radio is so that the radio appears in the void. The naming's definitely confusing, `Radio` is the void model, `RadioEnt` is the snatchable "real" one.)

**Comment by Commenter25** [comment]
That makes more sense. I had gotten confused when I saw the snatched radio in the void, then I just tried having it remove the other one and it worked, so I shrugged and moved on. My impression was that it was needed as a proxy for the music to be controlled after snatching, so it could be stopped when the trolley starts. But since this stops it beforehand anyway, that's irrelevant.

**Review by SageJFox — APPROVED** [review]
Didn't wanna approve this without checking some stuff, but checked was some stuff, and shouldn't be an issue.

File: gamemodes/jazztronauts/entities/entities/jazz_bus_explore/init.lua [hunk 1]
```diff
@@ -215,6 +215,11 @@ function ENT:AttachRadio(pos, ang)
 
 	-- Attach a looping audiozone
 	self.RadioMusic = CreateSound(ent, self.RadioMusicName)
+	hook.Add("EntityRemoved", "CheckTheRadio", function(removed)
```
Comment (Foohy) [inline gamemodes/jazztronauts/entities/entities/jazz_bus_explore/init.lua:218]: Really minor, and I realized I should've written this on the previous PRs, but for gmod hooks I try to use a a "jazz" prefix to keep it unique. I wouldn't be surprised if there's some other rascal out there with a radio mod and a hook called CheckTheRadio.

Comment (SageJFox) [reply gamemodes/jazztronauts/entities/entities/jazz_bus_explore/init.lua:218]: But what if they have a jazz radio mod?

Comment (Commenter25) [reply gamemodes/jazztronauts/entities/entities/jazz_bus_explore/init.lua]: Ah, you're absolutely right! Thankfully the others are fairly specific, only quickly running and with a rather specific name and hook, but this one is definitely far too vague. I've changed it to "JazzBusRadioCheck", which I don't think anything will conflict with. If we find an addon with a jazzy bus that lets you check the radio, then we'll deal with it then ;D

**Review by SageJFox — APPROVED** [review]

**Comment by Foohy** [comment]
Thank you kindly!!

**Pull request merged**